action/protocol/staking/validations.go

```diff
@@ -66,7 +70,11 @@ func (p *Protocol) validateCandidateRegister(ctx context.Context, act *action.Ca
 	}
 
 	if act.Amount().Cmp(p.config.RegistrationConsts.MinSelfStake) < 0 {
-		return errors.Wrap(action.ErrInvalidAmount, "self staking amount is not valid")
+		featureCtx := protocol.MustGetFeatureCtx(ctx)
+		if featureCtx.CandidateRegisterMustWithStake || act.Amount().Cmp(big.NewInt(0)) != 0 {
```

Comment:
`if featureCtx.CandidateRegisterMustWithStake` always return error?

Reply:
yes, only `featureCtx.CandidateRegisterMustWithStake==false && act.Amount()==0` is valid if `act.Amount() < MinSelfStake`

Reply:
!featureCtx.CandidateRegisterMustWithStake  && act.Amount()==0 is easier to read than featureCtx.CandidateRegisterMustWithStake || act.Amount().Cmp(big.NewInt(0)) != 0